---
title: "Figure3E_F"
author: "TÃ¼lay Karakulak"
date: "2024-03-15"
output: html_document
---

# Load Libraries
```{r loadLibraries}
library(Seurat)
library(dplyr)
library(ggplot2)
library(knitr)
library(rmarkdown)
library(RColorBrewer)
library(viridis)
set.seed(12345)
```


```{r readIsoforms}
ccRCC_Marker_Annotations <- read.table( '/Users/tulaykarakulak/Documents/PhD/Projects/ccRCC_Sc_lrSequencing/070923_Final/PaperRFiles/ccRCC_CellType.tsv', sep='\t', header = TRUE)
```

```{r readSeuratFiles}
seurat_annotations_ccRCC_2 <- read.csv('/Users/tulaykarakulak/Documents/GitHub/ccRCC_scLongRead/InputFiles/ccRCC2/scisoseq.seurat_info/scisoseq.annotated.info.csv', header = TRUE, sep='\t')

seurat_annotations_ccRCC_4 <- read.csv('/Users/tulaykarakulak/Documents/GitHub/ccRCC_scLongRead/InputFiles/ccRCC4/scisoseq.seurat_info/scisoseq.annotated.info.csv', header = TRUE, sep='\t')

seurat_annotations_ccRCC_5 <- read.csv('/Users/tulaykarakulak/Documents/GitHub/ccRCC_scLongRead/InputFiles/ccRCC5/scisoseq.seurat_info/scisoseq.annotated.info.csv', header = TRUE, sep='\t')
```

```{r makeDF}
# make a dataframe with Barcodes and sample 
ccRCC_Marker_Annotations$BC <- sapply(strsplit(ccRCC_Marker_Annotations$CellBarcode, "-"), function(x) x[1])
head(ccRCC_Marker_Annotations)
```

# analyse all cells
```{r assignCells}
seurat_annotations_ccRCC_2$Status <- ifelse(seurat_annotations_ccRCC_2$BC %in%  ccRCC_Marker_Annotations[ccRCC_Marker_Annotations$Sample == 'ccRCC2' & ccRCC_Marker_Annotations$CellType == 'ccRCC', 'BC'], 'ccRCC', 'non-ccRCC')

seurat_annotations_ccRCC_4$Status <- ifelse(seurat_annotations_ccRCC_4$BC %in%  ccRCC_Marker_Annotations[ccRCC_Marker_Annotations$Sample == 'ccRCC4' & ccRCC_Marker_Annotations$CellType == 'ccRCC', 'BC'], 'ccRCC', 'non-ccRCC')
seurat_annotations_ccRCC_4 <- seurat_annotations_ccRCC_4[seurat_annotations_ccRCC_4$BC %in% ccRCC_Marker_Annotations$BC, ]

seurat_annotations_ccRCC_5$Status <- ifelse(seurat_annotations_ccRCC_5$BC %in%  ccRCC_Marker_Annotations[ccRCC_Marker_Annotations$Sample == 'ccRCC5' & ccRCC_Marker_Annotations$CellType == 'ccRCC', 'BC'], 'ccRCC', 'non-ccRCC')
```


```{r distributionPlot}
BC_CellTypes_ccRCC2 <- seurat_annotations_ccRCC_2[, c(2, 4, 5, 6, 11, 13)] %>% dplyr::distinct()
BC_CellTypes_ccRCC4 <- seurat_annotations_ccRCC_4[, c(2, 4, 5, 6, 11, 13)] %>% dplyr::distinct()
BC_CellTypes_ccRCC5 <- seurat_annotations_ccRCC_5[, c(2, 4, 5, 6, 11, 13)] %>% dplyr::distinct()

BC_CellTypes_ccRCC2$Group <- 'ccRCC2'
BC_CellTypes_ccRCC4$Group <- 'ccRCC4'
BC_CellTypes_ccRCC5$Group <- 'ccRCC5'
```


```{r plot}
BC_CellTypes_AllccRCC <- rbind(BC_CellTypes_ccRCC2,  BC_CellTypes_ccRCC4, BC_CellTypes_ccRCC5)

BC_CellTypes_AllccRCC$category[BC_CellTypes_AllccRCC$category %in% c('fusion', 'genic', 'moreJunctions', 'NA', 'antisense', 'intergenic')]  <- 'Other'
```


```{r readSeuratObjects}
# Cancer
ccRCC_2.data_isoform <- Read10X(data.dir = "/Users/tulaykarakulak/Documents/GitHub/ccRCC_scLongRead/InputFiles/ccRCC2/scisoseq.seurat_info/isoforms_seurat/", gene.column = 1)
# Initialize the Seurat object with the raw (non-normalized data).
ccRCC_2_isoform <- CreateSeuratObject(counts = ccRCC_2.data_isoform, project = "ccRCC_2", min.cells = 3, min.features = 100)

ccRCC_3.data_isoform <- Read10X(data.dir = "/Users/tulaykarakulak/Documents/GitHub/ccRCC_scLongRead/InputFiles/ccRCC3/scisoseq.seurat_info/isoforms_seurat/", gene.column = 1)
# Initialize the Seurat object with the raw (non-normalized data).
ccRCC_3_isoform <- CreateSeuratObject(counts = ccRCC_3.data_isoform, project = "ccRCC_3", min.cells = 3, min.features = 100)

ccRCC_4.data_isoform <- Read10X(data.dir = "/Users/tulaykarakulak/Documents/GitHub/ccRCC_scLongRead/InputFiles/ccRCC4/scisoseq.seurat_info/isoforms_seurat/", gene.column = 1)
# Initialize the Seurat object with the raw (non-normalized data).
ccRCC_4_isoform <- CreateSeuratObject(counts = ccRCC_4.data_isoform, project = "ccRCC_4", min.cells = 3, min.features = 100)

ccRCC_5.data_isoform <- Read10X(data.dir = "/Users/tulaykarakulak/Documents/GitHub/ccRCC_scLongRead/InputFiles/ccRCC5/scisoseq.seurat_info/isoforms_seurat/", gene.column = 1)
# Initialize the Seurat object with the raw (non-normalized data).
ccRCC_5_isoform <- CreateSeuratObject(counts = ccRCC_5.data_isoform, project = "ccRCC_5", min.cells = 3, min.features = 100)

healthy.data_isoform <- Read10X(data.dir = "/Users/tulaykarakulak/Documents/GitHub/ccRCC_scLongRead/InputFiles/Normal/scisoseq.seurat_info/isoforms_seurat/", gene.column = 1)
healthy_isoform <- CreateSeuratObject(counts = healthy.data_isoform, project = "healthy", min.cells = 3, min.features = 100)
```


```{r selectIsoforms}
ccRCC2_Isoforms <- sapply(strsplit(rownames(ccRCC_2_isoform), ":"),`[`, 1)
ccRCC4_Isoforms <- sapply(strsplit(rownames(ccRCC_4_isoform), ":"),`[`, 1)
ccRCC5_Isoforms <- sapply(strsplit(rownames(ccRCC_5_isoform), ":"),`[`, 1)

BC_CellTypes_ccRCC2_seurat <- BC_CellTypes_ccRCC2[BC_CellTypes_ccRCC2$pbid %in% ccRCC2_Isoforms & BC_CellTypes_ccRCC2$BC %in% sapply(strsplit(colnames(ccRCC_2_isoform), "-"),`[`, 1),]
BC_CellTypes_ccRCC4_seurat <- BC_CellTypes_ccRCC4[BC_CellTypes_ccRCC4$pbid %in% ccRCC4_Isoforms  & BC_CellTypes_ccRCC4$BC %in% sapply(strsplit(colnames(ccRCC_4_isoform), "-"),`[`, 1),]
BC_CellTypes_ccRCC5_seurat <- BC_CellTypes_ccRCC5[BC_CellTypes_ccRCC5$pbid %in% ccRCC5_Isoforms  & BC_CellTypes_ccRCC5$BC %in% sapply(strsplit(colnames(ccRCC_5_isoform), "-"),`[`, 1),]

BC_CellTypes_ccRCC2_seurat$Group <- 'ccRCC2'
BC_CellTypes_ccRCC4_seurat$Group <- 'ccRCC4'
BC_CellTypes_ccRCC5_seurat$Group <- 'ccRCC5'


BC_CellTypes_AllccRCC_seurat <- rbind(BC_CellTypes_ccRCC2_seurat, BC_CellTypes_ccRCC4_seurat, BC_CellTypes_ccRCC5_seurat)

BC_CellTypes_AllccRCC_seurat$category[BC_CellTypes_AllccRCC_seurat$category %in% c('fusion', 'genic', 'moreJunctions', 'NA', 'antisense', 'intergenic')]  <- 'Other'
```


```{r SpecificIsoformCategories}
# PBids in ccRCC+ and not in ccRCC-
pbids_ccRCC4_plus_only <- BC_CellTypes_ccRCC4_seurat[ BC_CellTypes_ccRCC4_seurat$Status == 'ccRCC' & !BC_CellTypes_ccRCC4_seurat$pbid %in% BC_CellTypes_ccRCC4_seurat[BC_CellTypes_ccRCC4_seurat$Status == 'non-ccRCC', 'pbid'],]

pbids_ccRCC5_plus_only <- BC_CellTypes_ccRCC5_seurat[ BC_CellTypes_ccRCC5_seurat$Status == 'ccRCC' & !BC_CellTypes_ccRCC5_seurat$pbid %in% BC_CellTypes_ccRCC5_seurat[BC_CellTypes_ccRCC5_seurat$Status == 'non-ccRCC', 'pbid'],]

pbids_ccRCC2_plus_only <- BC_CellTypes_ccRCC2_seurat[ BC_CellTypes_ccRCC2_seurat$Status == 'ccRCC' & !BC_CellTypes_ccRCC2_seurat$pbid %in% BC_CellTypes_ccRCC2_seurat[BC_CellTypes_ccRCC2_seurat$Status == 'non-ccRCC', 'pbid'],]

# PBids in ccRCC- and not in ccRCC+
pbids_ccRCC4_neg_only <- BC_CellTypes_ccRCC4_seurat[ BC_CellTypes_ccRCC4_seurat$Status == 'non-ccRCC' & !BC_CellTypes_ccRCC4_seurat$pbid %in% BC_CellTypes_ccRCC4_seurat[BC_CellTypes_ccRCC4_seurat$Status == 'ccRCC', 'pbid'],]

pbids_ccRCC5_neg_only <- BC_CellTypes_ccRCC5_seurat[ BC_CellTypes_ccRCC5_seurat$Status == 'non-ccRCC' & !BC_CellTypes_ccRCC5_seurat$pbid %in% BC_CellTypes_ccRCC5_seurat[BC_CellTypes_ccRCC5_seurat$Status == 'ccRCC', 'pbid'],]

pbids_ccRCC2_neg_only <- BC_CellTypes_ccRCC2_seurat[ BC_CellTypes_ccRCC2_seurat$Status == 'non-ccRCC' & !BC_CellTypes_ccRCC2_seurat$pbid %in% BC_CellTypes_ccRCC2_seurat[BC_CellTypes_ccRCC2_seurat$Status == 'ccRCC', 'pbid'],]

```

## Figure 3E
```{r isoformTypes}
BC_CellTypes_AllccRCC_seurat_neg_pos <- rbind(
  pbids_ccRCC2_neg_only, pbids_ccRCC2_plus_only, pbids_ccRCC4_neg_only,pbids_ccRCC4_plus_only,  pbids_ccRCC5_neg_only,pbids_ccRCC5_plus_only)

BC_CellTypes_AllccRCC_seurat_neg_pos$category[BC_CellTypes_AllccRCC_seurat_neg_pos$category %in% c('fusion', 'genic', 'moreJunctions', 'NA', 'antisense', 'intergenic')]  <- 'Other'

BC_CellTypes_AllccRCC_seurat_neg_pos_seurat <- BC_CellTypes_AllccRCC_seurat_neg_pos %>% dplyr::select(pbid, gene, category, Status, Group)  %>% dplyr::distinct() %>% group_by(Group, Status, category) %>%
    summarise(count = n()) %>%
    mutate(total = sum(count)) %>%
    mutate(proportion = count/total) %>% ungroup() 

cellTypesPro_seurat_pos_neg <- ggplot(BC_CellTypes_AllccRCC_seurat_neg_pos_seurat, aes(x = proportion, y = Status, fill = category)) +
  geom_bar(stat = "identity", position = "stack") +
  labs(y = "CellTypes", x = "proportion", title = "Proportion of Status-specific Transcript Categories across ccRCC Status") +
  scale_fill_manual(values = color_palette_inferno) + 
  theme_bw() +
  facet_wrap(~Group, scales="free_x") +
  theme(legend.position="right")

cellTypesPro_seurat_pos_neg
ggsave("Figure3E.png", plot = cellTypesPro_seurat_pos_neg, bg = "transparent", width = 8, height = 2)
```
# Supplementary Fig 5A and 5B, and Supplementary Table 2
```{r VennDiagram}
# install.packages("ggVennDiagram")
library(ggVennDiagram)

# List of items
ccRCC_Pos_Overlaps <- list(ccRCC2 = unique(pbids_ccRCC2_plus_only[pbids_ccRCC2_plus_only$category %in% c('novel_in_catalog', 'novel_not_in_catalog'),]$gene), ccRCC4 = unique(pbids_ccRCC4_plus_only[pbids_ccRCC4_plus_only$category %in% c('novel_in_catalog', 'novel_not_in_catalog'),]$gene), ccRCC5 = unique(pbids_ccRCC5_plus_only[pbids_ccRCC5_plus_only$category %in% c('novel_in_catalog', 'novel_not_in_catalog'),]$gene))

# 2D Venn diagram
ggVennDiagram(ccRCC_Pos_Overlaps)

# List of items
ccRCC_Neg_Overlaps <- list(ccRCC2 = unique(pbids_ccRCC2_neg_only[pbids_ccRCC2_neg_only$category %in% c('novel_in_catalog', 'novel_not_in_catalog'),]$gene), ccRCC4 = unique(pbids_ccRCC4_neg_only[pbids_ccRCC4_neg_only$category %in% c('novel_in_catalog', 'novel_not_in_catalog'),]$gene), ccRCC5 = unique(pbids_ccRCC5_neg_only[pbids_ccRCC5_neg_only$category %in% c('novel_in_catalog', 'novel_not_in_catalog'),]$gene))

# 2D Venn diagram
ggVennDiagram(ccRCC_Neg_Overlaps)

supp_5a <- ggVennDiagram(ccRCC_Pos_Overlaps)
supp_5b <- ggVennDiagram(ccRCC_Neg_Overlaps)
ggsave("SupplementaryFigure5A.png", plot = supp_5a, bg = "transparent", width = 5, height = 5)
ggsave("SupplementaryFigure5B.png", plot = supp_5b, bg = "transparent", width = 5, height = 5)

# SupplementaryTable2
SupplementaryTable2 <- rbind(pbids_ccRCC2_plus_only[pbids_ccRCC2_plus_only$category %in% c('novel_in_catalog', 'novel_not_in_catalog'),],
pbids_ccRCC4_plus_only[pbids_ccRCC4_plus_only$category %in% c('novel_in_catalog', 'novel_not_in_catalog'),],
pbids_ccRCC5_plus_only[pbids_ccRCC5_plus_only$category %in% c('novel_in_catalog', 'novel_not_in_catalog'),])

SupplementaryTable2_NumberOfCells <- SupplementaryTable2 %>% dplyr::group_by(pbid, transcript, gene, category, Status, Group) %>% dplyr::summarise(NumberOfCells = n()) %>% dplyr::arrange(desc(NumberOfCells))

colnames(SupplementaryTable2_NumberOfCells) <- c('isoform', 'transcript', 'gene', 'structural_category', 'Cell Type', 'Sample ID', 'Number Of Cells')

#write.table(SupplementaryTable2_NumberOfCells, 'Output/Supplementary_Table_2.tsv', quote = FALSE, row.names = FALSE, sep = '\t')
```


length(intersect(unique(pbids_ccRCC2_plus_only[pbids_ccRCC2_plus_only$category %in% c('novel_in_catalog', 'novel_not_in_catalog'),'gene']), unique(pbids_ccRCC5_plus_only[pbids_ccRCC5_plus_only$category %in% c('novel_in_catalog', 'novel_not_in_catalog'),'gene'])))

unique(pbids_ccRCC2_plus_only[pbids_ccRCC2_plus_only$category %in% c('novel_in_catalog', 'novel_not_in_catalog') & pbids_ccRCC2_plus_only$gene %in% pbids_ccRCC2_plus_only[pbids_ccRCC2_plus_only$category %in% c('novel_in_catalog', 'novel_not_in_catalog'), 'gene'],'gene'])

length(unique(pbids_ccRCC2_plus_only$gene))
[1] 13927
length(unique(pbids_ccRCC5_plus_only$gene))
[1] 2812

```{r checkCellCounts}
intersect2_5 <- intersect(unique(pbids_ccRCC2_plus_only[pbids_ccRCC2_plus_only$category %in% c('novel_in_catalog', 'novel_not_in_catalog'),'gene']), unique(pbids_ccRCC5_plus_only[pbids_ccRCC5_plus_only$category %in% c('novel_in_catalog', 'novel_not_in_catalog'),'gene']))

pbids_ccRCC2_plus_only_Novel_CellCount <- pbids_ccRCC2_plus_only[pbids_ccRCC2_plus_only$gene %in% intersect2_5 & pbids_ccRCC2_plus_only$category %in% c('novel_in_catalog', 'novel_not_in_catalog'), ] %>% dplyr::select(pbid, gene, BC, Status, Group)  %>% dplyr::distinct() %>% group_by(pbid, gene, Group) %>%  dplyr::summarise(countOfCell = n()) %>% dplyr::arrange(desc(countOfCell))

pbids_ccRCC5_plus_only_Novel_CellCount <- pbids_ccRCC5_plus_only[pbids_ccRCC5_plus_only$gene %in% intersect2_5 & pbids_ccRCC5_plus_only$category %in% c('novel_in_catalog', 'novel_not_in_catalog'), ] %>% dplyr::select(pbid, gene, BC, Status, Group) %>% dplyr::distinct() %>% group_by(pbid, gene, Group) %>% dplyr::summarise(countOfCell = n()) %>% dplyr::arrange(desc(countOfCell))

pbids_ccRCC4_plus_only_Novel_CellCount <- pbids_ccRCC4_plus_only[pbids_ccRCC4_plus_only$gene %in% intersect2_5 & pbids_ccRCC4_plus_only$category %in% c('novel_in_catalog', 'novel_not_in_catalog'), ] %>% dplyr::select(pbid, gene, BC, Status, Group) %>% dplyr::distinct() %>% group_by(pbid, gene, Group) %>% dplyr::summarise(countOfCell = n()) %>% dplyr::arrange(desc(countOfCell))
```

## Figure 3G
```{r clusterProfiler}
library(clusterProfiler)
library(msigdbr)
m_df <- msigdbr(species = "Homo sapiens")  %>% 
  dplyr::select(gs_name, entrez_gene)
m_t2g <- msigdbr(species = "Homo sapiens", category = "C6") %>% 
  dplyr::select(gs_name, entrez_gene)

m_H <- msigdbr(species = "Homo sapiens", category = "H") %>% 
  dplyr::select(gs_name, entrez_gene)

m_c4 <- msigdbr(species = "Homo sapiens", category = "C4") %>% 
  dplyr::select(gs_name, entrez_gene)


eg = bitr(intersect2_5, fromType="SYMBOL", toType="ENTREZID", OrgDb="org.Hs.eg.db")

library(enrichplot)
WP <- enrichWP(eg$ENTREZID, organism = "Homo sapiens") 
wp_result <- dotplot(WP, showCategory=10)
wp_result
ggsave("Figure3G.png", plot = wp_result, bg = "transparent", width = 8, height = 10)
```








