---
title: "Figure 3"
author: "TÃ¼lay Karakulak"
output: 
  html_document:
    toc: true
    toc_depth: 2
---

# Load Libraries
```{r loadLibraries}
library(Seurat)
library(dplyr)
library(ggplot2)
library(knitr)
library(rmarkdown)
library(RColorBrewer)
library(viridis)
set.seed(12345)
```

## Read Seurat Data
```{r loadData, message=FALSE}
create_seurat_object <- function(data_dir, project_name) {
  data_gene <- Read10X(data.dir = data_dir, gene.column = 2)
  CreateSeuratObject(counts = data_gene, project = project_name, min.cells = 3, min.features = 100)
}

datasets <- list(
  healthy = "/Users/tulaykarakulak/Documents/GitHub/ccRCC_scLongRead/InputFiles/Normal/scisoseq.seurat_info/genes_seurat/",
  ccRCC_2 = "/Users/tulaykarakulak/Documents/GitHub/ccRCC_scLongRead/InputFiles/ccRCC2/scisoseq.seurat_info/genes_seurat/",
  ccRCC_3 = "/Users/tulaykarakulak/Documents/GitHub/ccRCC_scLongRead/InputFiles/ccRCC3/scisoseq.seurat_info/genes_seurat/",
  ccRCC_4 = "/Users/tulaykarakulak/Documents/GitHub/ccRCC_scLongRead/InputFiles/ccRCC4/scisoseq.seurat_info/genes_seurat/",
  ccRCC_5 = "/Users/tulaykarakulak/Documents/GitHub/ccRCC_scLongRead/InputFiles/ccRCC5/scisoseq.seurat_info/genes_seurat/"
)

kidney.combined <- list()

for (name in names(datasets)) {
  kidney.combined[[name]] <- create_seurat_object(datasets[[name]], name)
}
```

## SCTransform
Tutorial used for batch correction: https://bioinformatics-core-shared-training.github.io/SingleCell_RNASeq_Jan23/UnivCambridge_ScRnaSeqIntro_Base/Markdowns/101-seurat_part2.html by changing variable.features.n to 3000, and nfeatures to 2000, and adding a sample.tree to integrate ccRCC samples first, and then the healthy sample (integration order: ccRCC2-ccRCC5-ccRCC4-ccRCC3-healthy)

```{r integrateCells, message=FALSE, max.height='100px', attr.output='.numberLines', warning=FALSE}
# loop through the individual samples, normalize and scale them
kidney.combined <- lapply(X = kidney.combined, FUN = function(x) {
    invisible(capture.output(
    x <- SCTransform(x,  verbose = TRUE, variable.features.n = 3000)
    ))
  return(x)
})
# integrate all samples 
features <- SelectIntegrationFeatures(object.list = kidney.combined, nfeatures=2000)
kidney.combined_anchors <- FindIntegrationAnchors(object.list = kidney.combined, anchor.features = features)
kidney.combined <- IntegrateData(anchorset = kidney.combined_anchors, normalization.method = "SCT", 
                                 sample.tree = matrix(c(-2, 1, 2, 3, -5, -4, -3, -1), ncol = 2))
DefaultAssay(kidney.combined)

```

```{r SCTransformData}
#library(scGate) - # marker detection in the individual data

kidney.combined2 <- list()

for (name in names(datasets)) {
  kidney.combined2[[name]] <- create_seurat_object(datasets[[name]], name)
}

kidney.combined2 <- lapply(X = kidney.combined2, FUN = function(x) {
    invisible(capture.output(
    x <- SCTransform(x,  verbose = TRUE, variable.features.n = 3000)
    ))
  return(x)
})

```


```{r scGate}
library(scGate)
my_scGate_model <- scGate::gating_model(name = "ccRCC", signature = c("CA9"))  
ccRCC2_kidney <- ScaleData(kidney.combined2$ccRCC_2, verbose = FALSE)
ccRCC2_kidney <- RunPCA(ccRCC2_kidney, npcs = 20, verbose = FALSE)
ccRCC2_kidney <- RunUMAP(ccRCC2_kidney, reduction = "pca", dims = 1:16)
ccRCC_2_scgate <- scGate(data = ccRCC2_kidney, model = my_scGate_model)
DimPlot(ccRCC_2_scgate, group.by = "is.pure")

#scGate it!
ccRCC4_kidney <- ScaleData(kidney.combined2$ccRCC_4, verbose = FALSE)
ccRCC4_kidney <- RunPCA(ccRCC4_kidney, npcs = 20, verbose = FALSE)
ccRCC4_kidney <- RunUMAP(ccRCC4_kidney, reduction = "pca", dims = 1:16)
ccRCC_4_scgate <- scGate(data = ccRCC4_kidney, model = my_scGate_model)
DimPlot(ccRCC_4_scgate, group.by = "is.pure")

ccRCC5_kidney <- ScaleData(kidney.combined2$ccRCC_5, verbose = FALSE)
ccRCC5_kidney <- RunPCA(ccRCC5_kidney, npcs = 20, verbose = FALSE)
ccRCC5_kidney <- RunUMAP(ccRCC5_kidney, reduction = "pca", dims = 1:16)
ccRCC_5_scgate <- scGate(data = ccRCC5_kidney, model = my_scGate_model)
DimPlot(ccRCC_5_scgate, group.by = "is.pure")
```


```{r CA9PosAndNeg}
Cell_Sample_Type_ccRCC2 <- as.data.frame(Idents(ccRCC_2_scgate))
Cell_Sample_Type_ccRCC2$BC <- rownames(Cell_Sample_Type_ccRCC2)
colnames(Cell_Sample_Type_ccRCC2) <- c('CellType', 'CellBarcode')
Cell_Sample_Type_ccRCC2$CellType <- ifelse(Cell_Sample_Type_ccRCC2$CellType == 'Pure', 'ccRCC', 'non-ccRCC')
Cell_Sample_Type_ccRCC2$Sample <- 'ccRCC2'

Cell_Sample_Type_ccRCC4 <- as.data.frame(Idents(ccRCC_4_scgate))
Cell_Sample_Type_ccRCC4$BC <- rownames(Cell_Sample_Type_ccRCC4)
colnames(Cell_Sample_Type_ccRCC4) <- c('CellType', 'CellBarcode')
Cell_Sample_Type_ccRCC4$CellType <- ifelse(Cell_Sample_Type_ccRCC4$CellType == 'Pure', 'ccRCC', 'non-ccRCC')
Cell_Sample_Type_ccRCC4$Sample <- 'ccRCC4'

Cell_Sample_Type_ccRCC5 <- as.data.frame(Idents(ccRCC_5_scgate))
Cell_Sample_Type_ccRCC5$BC <- rownames(Cell_Sample_Type_ccRCC5)
colnames(Cell_Sample_Type_ccRCC5) <- c('CellType', 'CellBarcode')
Cell_Sample_Type_ccRCC5$CellType <- ifelse(Cell_Sample_Type_ccRCC5$CellType == 'Pure', 'ccRCC', 'non-ccRCC')
Cell_Sample_Type_ccRCC5$Sample <- 'ccRCC5'

Cell_Sample_Type <- rbind(Cell_Sample_Type_ccRCC5, Cell_Sample_Type_ccRCC4, Cell_Sample_Type_ccRCC2)

# Save Cell Types
#write.table(Cell_Sample_Type, '/Users/tulaykarakulak/Documents/PhD/Projects/ccRCC_Sc_lrSequencing/070923_Final/PaperRFiles/ccRCC_CellType.tsv', quote = FALSE, row.names = FALSE, sep='\t')
```

# Scale and PCA 
```{r ScaleandPCA, message=FALSE}
DefaultAssay(kidney.combined) <- 'integrated' 
kidney.combined <- ScaleData(kidney.combined, verbose = FALSE)
kidney.combined <- RunPCA(kidney.combined, npcs = 20, verbose = FALSE)
ElbowPlot(kidney.combined)
```

# UMAP after batch correction
```{r runUmap, message=FALSE}
kidney.combined <- RunUMAP(kidney.combined, reduction = "pca", dims = 1:16)
DimPlot(kidney.combined, reduction = "umap", group.by = "orig.ident")
```

# Distribution of Cells Across Clusters
```{r DistributionOfCells, message=FALSE}
kidney.combined <- FindNeighbors(kidney.combined, dims = 1:16)
kidney.combined <- FindClusters(kidney.combined, resolution = 0.8)
table(Cluster = kidney.combined$integrated_snn_res.0.8, Batch = kidney.combined$orig.ident)
```

```{r UMAP_Cluster, message=FALSE}
DimPlot(kidney.combined, reduction = "umap", label = TRUE)
```

# Find Markers
```{r findMarkers, message=FALSE}
# find specific markers for all clusters
# change default assay
DefaultAssay(kidney.combined) <- "SCT" # As the batch-corrected read counts are not suitable for looking at gene expression. the SCT-normalised data is used when examining marker expression.
kidney.combined <- PrepSCTFindMarkers(kidney.combined)
call_markers_all <- FindAllMarkers(kidney.combined, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)

# get the top 5 hits for each cluster
call_markers_all %>%
  group_by(cluster) %>%
  slice_max(n = 5, order_by = avg_log2FC)

cell_markers <- call_markers_all
```

## Supplementary Fig 4A
```{r SCpubrDimPlot}
library(SCpubr)
figure3A <- SCpubr::do_DimPlot(sample = kidney.combined, label = TRUE, plot.axes = TRUE, label.box = TRUE)
ggsave("Figure3A.png", plot = figure3A, bg = "transparent", width = 8, height = 5)
figure3A
figure4A_category <- SCpubr::do_DimPlot(sample = kidney.combined, label = TRUE, plot.axes = TRUE, label.box = TRUE, split.by = 'orig.ident')
ggsave("Supplementary_Figure4A.png", plot = figure4A_category, bg = "transparent", width = 8, height = 5)
figure4A_category
```

## Supplementary Fig 4B
```{r EpithelialMarkers}
DefaultAssay(kidney.combined) <- 'SCT'
SCpubr::do_FeaturePlot(sample = kidney.combined, 
                             features = "EPCAM",
                             plot.title = "Expression of Epithelial Marker: EPCAM", 
                             reduction = "umap")

DefaultAssay(kidney.combined) <- 'SCT'
SCpubr::do_FeaturePlot(sample = kidney.combined, 
                             features = "PAX8",
                             plot.title = "Expression of Epithelial Marker: PAX8", 
                             reduction = "umap")


epcam <- SCpubr::do_FeaturePlot(sample = kidney.combined, 
                             features = "EPCAM",
                             plot.title = "Expression of Epithelial Marker: EPCAM", 
                             reduction = "umap", split.by = 'orig.ident')

pax8 <- SCpubr::do_FeaturePlot(sample = kidney.combined, 
                             features = "PAX8",
                             plot.title = "Expression of Epithelial Marker: PAX8", 
                             reduction = "umap", split.by = 'orig.ident')

ggsave("Supplementary_Figure4B_1.png", plot = epcam, bg = "transparent", width = 8, height = 5)
ggsave("Supplementary_Figure4B_2.png", plot = pax8, bg = "transparent", width = 8, height = 5)
```
## Supplementary Fig 4C
```{r ccRCCMarkers}
ca9 <- SCpubr::do_FeaturePlot(sample = kidney.combined, 
                             features = "CA9",
                             plot.title = "Expression of ccRCC Marker: CA9", 
                             reduction = "umap", split.by = 'orig.ident')


ANGPTL4 <- SCpubr::do_FeaturePlot(sample = kidney.combined, 
                             features = "ANGPTL4",
                             plot.title = "Expression of ccRCC Marker: ANGPTL4", 
                             reduction = "umap", split.by = 'orig.ident')

NDUFA4L2 <- SCpubr::do_FeaturePlot(sample = kidney.combined, 
                             features = "NDUFA4L2",
                             plot.title = "Expression of ccRCC Marker: NDUFA4L2", 
                             reduction = "umap", split.by = 'orig.ident')


VIM <- SCpubr::do_FeaturePlot(sample = kidney.combined, 
                             features = "VIM",
                             plot.title = "Expression of ccRCC Marker: VIM", 
                             reduction = "umap", split.by = 'orig.ident')

ggsave("Supplementary_Figure4C_1.png", plot = ca9, bg = "transparent", width = 8, height = 5)
ggsave("Supplementary_Figure4C_2.png", plot = ANGPTL4, bg = "transparent", width = 8, height = 5)
```


# Supplementary Figure 4D
```{r proximalTubuleMarkers}
PTC_GGT1 <- SCpubr::do_FeaturePlot(sample = kidney.combined, 
                             features = "GGT1",
                             plot.title = "Expression of PTC Marker: GGT1", 
                             reduction = "umap", split.by = 'orig.ident')


RIDA <- SCpubr::do_FeaturePlot(sample = kidney.combined, 
                             features = "RIDA",
                             plot.title = "Expression of PTC Marker: RIDA", 
                             reduction = "umap", split.by = 'orig.ident')

SLC22A3 <- SCpubr::do_FeaturePlot(sample = kidney.combined, 
                             features = "SLC22A3",
                             plot.title = "Expression of PTC Marker: RIDA", 
                             reduction = "umap", split.by = 'orig.ident')

ggsave("Supplementary_Figure4D_1.png", plot = PTC_GGT1, bg = "transparent", width = 8, height = 5)
ggsave("Supplementary_Figure4D_2.png", plot = RIDA, bg = "transparent", width = 8, height = 5)
```



# Select ccRCC+ and ccRCC- cells from Batch Corrected Object
```{r selectccRCCpositiveandNegativeCells}
# Cells expressing CA9
#allcells_CA9_high <- WhichCells(kidney.combined, expression = CA9 > 0, slot = 'data')

Cell_Sample_Type <- Cell_Sample_Type %>% mutate(ExtractedNumber = gsub("ccRCC", "", Sample),
           # Concatenate CellBarcode with the ExtractedNumber, separated by an underscore
           NewCellBarcode = paste(CellBarcode, ExtractedNumber, sep = "_"))

ccRCC_cells <- Cell_Sample_Type[Cell_Sample_Type$CellType == 'ccRCC', 'NewCellBarcode']

# Assign cells as ccRCC
Idents(kidney.combined, cells = ccRCC_cells) <- 'ccRCC'

# Assign remaining cells as non-ccRCC
all_cells <- colnames(kidney.combined)

remaining_cells <- setdiff(all_cells, ccRCC_cells)

Idents(kidney.combined, cells = remaining_cells) <- 'non-ccRCC'
```

> length(Idents(kidney.combined)[Idents(kidney.combined) == 'ccRCC'])
[1] 620
> length(Idents(kidney.combined)[Idents(kidney.combined) == 'non-ccRCC'])
[1] 1978


```{r saveCells}
# make a the dataframe
Cell_Sample_Type <- as.data.frame(Idents(kidney.combined))
Cell_Sample_Type$BC <- rownames(Cell_Sample_Type)
colnames(Cell_Sample_Type) <- c('CellType', 'CellBarcode')
Cell_Sample_Type2 <-  Cell_Sample_Type %>%
  # Separate the 'CellType' column into two columns: 'CellType' (without the suffix) and 'Number' (the suffix after '_')
  tidyr::separate(col = CellBarcode, into = c("CellBarcode", "Number"), sep = "_", remove = FALSE) %>%
  # Update 'CellBarcode' to concatenate 'ccRCC' with the 'Number', also add the underscore back
  dplyr::mutate(Sample = paste0("ccRCC_", Number)) %>%
  # Now, remove the 'Number' column as it's no longer needed
  dplyr::select(-Number)

Cell_Sample_Type2$Sample[Cell_Sample_Type2$Sample == 'ccRCC_1'] <- 'Healthy'
#write.table(Cell_Sample_Type2, 'Output/', quote = FALSE, row.names = FALSE, sep='\t')
```

## Figure 3C
## Cell distribution of samples across ccRCC+ and ccRCC-
```{r DistributionOfCellsinccRCCPosandNeg, message=FALSE}
table(Cluster = kidney.combined$ccRCCAnnotations, Batch = kidney.combined$orig.ident)


data <- data.frame(
  Cluster = c("ccRCC", "non-ccRCC"),
  ccRCC2 = c(361, 12),
  ccRCC3 = c(0, 310),
  ccRCC4 = c(42, 1048),
  ccRCC5 = c(217, 171),
  Normal = c(0, 437)
)

# Transform the data to long format
data_long <- pivot_longer(data, cols = -Cluster, names_to = "Batch", values_to = "Count")
library(ggsci)
# Create the bar plot
NumberOfCells <- ggplot(data_long, aes(x = Batch, y = Count, fill = Cluster)) +
  geom_bar(stat = "identity", position = position_dodge(width = 0.9)) +
  geom_text(aes(label = Count), vjust = -0.25, position = position_dodge(width = 0.9), size = 3.5) +
    theme_bw() +
  theme(text = element_text(size = 12),
        axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(title = "Number of ccRCC and non-ccRCC cells",
       x = "Samples", 
       y = "# of Cell") + scale_fill_npg()

ggsave("Figure3C_BarGraph.png", plot = NumberOfCells, bg = "transparent", width = 8, height = 5)

```

## Figure 3D
```{r FindMarkersBetweenccRCCPosandNeg}
kidney.combined <- PrepSCTFindMarkers(kidney.combined)
de_genes <- FindAllMarkers(kidney.combined, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25, recorrect_umi=FALSE)
```


```{r plotDEG}
# Heatmap
p_deg <- SCpubr::do_GroupwiseDEPlot(sample = kidney.combined,
                                de_genes = de_genes,
                                top_genes = 10)
p_deg
ggsave("Figure3D.png", plot = p_deg, bg = "transparent", width = 16, height = 5)
```




